service/courseService/courseService_test.go

```diff
@@ -123,3 +123,201 @@ func TestCourseService_GetCourse(t *testing.T) {
 		})
 	}
 }
+
+func TestCourseService_GetCoursesByMember(t *testing.T) {
+	t.Parallel()
+
+	var mockCourseRepo mock.CourseRepository
+	service := CourseService{CR: &mockCourseRepo}
+
+	course1 := eduboard.Course{ID: "1", Title: "Course 1", Members: []string{"1"}}
+
+	testCases := []struct {
+		name     string
+		input    string
+		error    bool
+		expected []eduboard.Course
+	}{
+		{"success", "1", false, []eduboard.Course{course1}},
+		{"error", "", true, []eduboard.Course{}},
+	}
+
+	mockCourseRepo.FindByMemberFn = func(member string) (error, []eduboard.Course) {
+		switch member {
+		case "1":
+			return nil, []eduboard.Course{course1}
+		default:
+			return errors.New("Error fetching courses"), []eduboard.Course{}
+		}
+	}
+
+	for _, v := range testCases {
+		t.Run(fmt.Sprintf("GetCoursesByMember %s", v.name), func(t *testing.T) {
```

Comment:
`go test -v` already prints the name of the testing function, in other cases this would not be printed anyways.